### PR TITLE
remove requirements(-*).txt and update docs

### PR DESCRIPTION
**Pull request by ankona**
- Delete requirements.txt, requirements-dev.txt, requirements-mypy.txt
- Remove requirements.txt from manifest.in
- Update installation instructions in docs to use setup.py

### CODE REVIEW

**Review by mellis13 — APPROVED** [review]
Thanks for making this long overdue change.  LGTM!

**Review by MattToast — CHANGES_REQUESTED** [review]
two small clarifications requested

File: doc/developer.rst [hunk 1]
```diff
@@ -73,8 +73,10 @@ SmartSim, users can run multiple testing commands with the developer Makefile
 
 .. note::
 
-  For the test to run, you must have the ``requirements-dev.txt`` dependencies
-  installed in your python environment.
+You must have the ``extras_require==dev`` dependencies installed in
```
[the remark this answers is not in the document]
Comment (ankona) [reply doc/developer.rst:76]: updated per request

File: doc/testing.rst [hunk 2]
```diff
@@ -23,12 +23,13 @@ level of the SmartSim directory::
...
                         WARNING: do not run test-full on shared systems.
 
-For the test to run, you must have the ``requirements-dev.txt``
-dependencies installed in your python environment.
+.. note::
 
-There are two ways to do this
-1. Install smartsim with ``dev`` extension ``pip install -e .[dev]``
-2. ``pip install -r requirements-dev.txt`` after you install smartsim.
+You must have the ``extras_require==dev`` dependencies installed in
```
[the remark this answers is not in the document]
Comment (ankona) [reply doc/testing.rst:28]: updated per request

**Review by MattToast — APPROVED** [review]
LGTM!!

**Pull request merged**